harness/tests/integration_cases/test_raft_flow_control.rs

```diff
@@ -175,3 +175,24 @@ fn test_msg_app_flow_control_recv_heartbeat() {
         r.read_messages();
     }
 }
+
+// #[test]
+// fn test_msg_app_flow_control_with_freeing_resources() {
+//     let l = default_logger();
+//     let mut r = new_test_raft(1, vec![1, 2], 5, 1, new_storage(), &l);
+//     r.become_candidate();
+//     r.become_leader();
+//
+//     // force the progress to be in replicate state
+//     r.mut_prs().get_mut(2).unwrap().become_replicate();
+//
+//     // send message from 2 to 1 for adding inflights
+//     for _ in 0..10 {
+//         r.step(new_message(1, 1, MessageType::MsgPropose, 1))
+//             .expect("");
+//         r.read_messages();
+//     }
+//
```

Comment:
After this line, inflights 2 (follower) is filled with 10 element, inflights 1 (leader) is empty, so it seems that we need to transfer leader from 2 to 1, then transfer 1 to 2 again, so peer 2 will call `free_mem` ?